### DISCUSSION TITLE
Filterbot Creation and Standards

### DISCUSSION THREAD

**Discussion by duubi-official** [NovaoTech/solaris#9] [Development]
I've been thinking that we will need a filterbot, but one that's better than scratch's filterbot. If anyone has any ideas for how it should work, what is deemed as inappropriate, ect., we can discuss them here.

**Comment by MikeDev101** [comment]
I think we should go with a blacklist-like system. We should offer the most freedom while offering protected chat/comment functionality.

**Reply by icarusinthewings** [reply]
^ This.

**Reply by MikeDev101** [reply]
There's a pretty good whitelist that exists on Scratch, it's Griffpatch's old whitelist. It contains plenty of words that are safe for use.  Anything not on the list should be reviewed. Of course, we have to write our own blacklist, however...

**Comment by dannyaguero** [comment]
For projects, sure, maybe Griffpatch's whitelist, but for other things like comments and such, can't there be a blacklist that contains bad words, but it's hidden to the public? maybe something that only admins access? this can lead to more freedom when it comes to words (and not censoring things that may seem bad, but aren't)

**Reply by MikeDev101** [reply]
As I mentioned in my suggestion, we should. As for the blacklist, it should be moderator/admin accessible/modifiable only. As for our whitelist, we should offer a suggest-and-review system for words that can be safely used.

**Comment by icarusinthewings** [comment]
If a comment gets flagged by the filterbot, it will be hidden to anyone except moderators and administrators who will be able to check the comment. We will also have a filterbot that will be ran on comments that get reported that is a bit more strict on words that could be used in a way to be mean to a person or group.
